sopel/plugins/rules.py

```diff
@@ -1017,30 +1048,30 @@ def from_callable_lazy(cls, settings, handler):
 
     def __init__(self,
                  regexes,
-                 plugin=None,
-                 label=None,
-                 priority=PRIORITY_MEDIUM,
-                 handler=None,
-                 events=None,
-                 ctcp=None,
-                 allow_bots=False,
-                 allow_echo=False,
-                 threaded=True,
-                 output_prefix=None,
-                 unblockable=False,
-                 user_rate_limit=0,
-                 channel_rate_limit=0,
-                 global_rate_limit=0,
-                 user_rate_message=None,
-                 channel_rate_message=None,
-                 global_rate_message=None,
-                 default_rate_message=None,
-                 usages=None,
-                 tests=None,
-                 doc=None):
+                 plugin: str = '',
+                 label: Optional[str] = None,
+                 priority: str = PRIORITY_MEDIUM,
+                 handler: Optional[Callable] = None,
+                 events: Optional[Iterable[str]] = None,
+                 ctcp: Optional[Iterable[Pattern]] = None,
+                 allow_bots: bool = False,
+                 allow_echo: bool = False,
+                 threaded: bool = True,
+                 output_prefix: Optional[str] = None,
+                 unblockable: bool = False,
+                 user_rate_limit: int = 0,
+                 channel_rate_limit: int = 0,
+                 global_rate_limit: int = 0,
+                 user_rate_message: Optional[str] = None,
+                 channel_rate_message: Optional[str] = None,
+                 global_rate_message: Optional[str] = None,
+                 default_rate_message: Optional[str] = None,
+                 usages: Optional[tuple] = None,
+                 tests: Optional[tuple] = None,
+                 doc: Optional[str] = None):
         # core
         self._regexes = regexes
-        self._plugin_name = plugin
+        self._plugin_name = plugin or ''
```

Comment:
Given ` plugin` is always a str now, there is no need for that (unless mypy complain? but that sounds weird):

```suggestion
        self._plugin_name = plugin
```

Note: I'm not even sure why I made `plugin`  optional in the first place.